includeResultMetadata types not changing return type?

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

8.1.1

### Node.js version

20.x

### MongoDB server version

6.x

### Typescript version (if applicable)

5.1.6

### Description

I tried upgrading to mongoose 8.x and changed rawResult: true to includeResultMetadata: true, but now typescript complains:
```
TS2339: Property value does not exist on type
FlattenMaps<IDBJobField> & Required<{   _id: ObjectId; }>

```

also it seems that the retutrn type is not changed depending on the value of includeResultMetadata
![image](https://github.com/Automattic/mongoose/assets/5757263/1c397a9e-b06a-4a3d-8a52-90d76227c79c)
it's the same as in the screenshot, regardless of this flag.

Furthermore; i found it kinda confusing with the mongodb docs, as they say that includeResultMetadata is true by default. Does mongoose set it to false then by default? Or is it actually true, but mongoose just returns the document and strips the rest away? Which would sound to me of a waste of some extra bytes?


error TS2339: Property 'value' does not exist on type 'FlattenMaps<IDBJobField> & Required<{ _id: ObjectId; }>'.
error TS2339: Property 'lastErrorObject' does not exist on type 'FlattenMaps<IDBJobField> & Required<{ _id: ObjectId; }>'.
error TS2339: Property 'value' does not exist on type 'FlattenMaps<IDBJobField> & Required<{ _id: ObjectId; }>'.


### Steps to Reproduce

use an example query for findOneAndupdate and play around with the includeResultMetadata flag

### Expected Behavior

should return the "ModifyResult" instead, like it did with rawResult: true

--- a/test/types/models.test.ts
+++ b/test/types/models.test.ts
@@ -17,7 +17,7 @@ import mongoose, {
 } from 'mongoose';
 import { expectAssignable, expectError, expectType } from 'tsd';
 import { AutoTypedSchemaType, autoTypedSchema } from './schema.test';
-import { UpdateOneModel, ChangeStreamInsertDocument, ObjectId } from 'mongodb';
+import { ModifyResult, UpdateOneModel, ChangeStreamInsertDocument, ObjectId } from 'mongodb';
 
 function rawDocSyntax(): void {
   interface ITest {
@@ -684,6 +684,9 @@ async function gh13705() {
 
   const findOneAndUpdateRes = await TestModel.findOneAndUpdate({}, {}, { lean: true });
   expectType<ExpectedLeanDoc | null>(findOneAndUpdateRes);
+
+  const findOneAndUpdateResWithMetadata = await TestModel.findOneAndUpdate({}, {}, { lean: true, includeResultMetadata: true });
+  expectAssignable<ModifyResult<ExpectedLeanDoc>>(findOneAndUpdateResWithMetadata);
 }
 
 async function gh13746() {

--- a/types/models.d.ts
+++ b/types/models.d.ts
@@ -593,6 +593,17 @@ declare module 'mongoose' {
 
     /** Creates a `findOneAndUpdate` query, filtering by the given `_id`. */
     findByIdAndUpdate<ResultDoc = THydratedDocumentType>(
+      filter: FilterQuery<TRawDocType>,
+      update: UpdateQuery<TRawDocType>,
+      options: QueryOptions<TRawDocType> & { includeResultMetadata: true, lean: true }
+    ): QueryWithHelpers<
+      ModifyResult<TRawDocType>,
+      ResultDoc,
+      TQueryHelpers,
+      TRawDocType,
+      'findOneAndUpdate'
+    >;
+    findByIdAndUpdate<ResultDoc = THydratedDocumentType>(
       id: mongodb.ObjectId | any,
       update: UpdateQuery<TRawDocType>,
       options: QueryOptions<TRawDocType> & { lean: true }
@@ -675,9 +686,9 @@ declare module 'mongoose' {
     findOneAndUpdate<ResultDoc = THydratedDocumentType>(
       filter: FilterQuery<TRawDocType>,
       update: UpdateQuery<TRawDocType>,
-      options: QueryOptions<TRawDocType> & { lean: true }
+      options: QueryOptions<TRawDocType> & { includeResultMetadata: true, lean: true }
     ): QueryWithHelpers<
-      GetLeanResultType<TRawDocType, TRawDocType, 'findOneAndUpdate'> | null,
+      ModifyResult<TRawDocType>,
       ResultDoc,
       TQueryHelpers,
       TRawDocType,
@@ -686,8 +697,14 @@ declare module 'mongoose' {
     findOneAndUpdate<ResultDoc = THydratedDocumentType>(
       filter: FilterQuery<TRawDocType>,
       update: UpdateQuery<TRawDocType>,
-      options: QueryOptions<TRawDocType> & { includeResultMetadata: true }
-    ): QueryWithHelpers<ModifyResult<ResultDoc>, ResultDoc, TQueryHelpers, TRawDocType, 'findOneAndUpdate'>;
+      options: QueryOptions<TRawDocType> & { lean: true }
+    ): QueryWithHelpers<
+      GetLeanResultType<TRawDocType, TRawDocType, 'findOneAndUpdate'> | null,
+      ResultDoc,
+      TQueryHelpers,
+      TRawDocType,
+      'findOneAndUpdate'
+    >;
     findOneAndUpdate<ResultDoc = THydratedDocumentType>(
       filter: FilterQuery<TRawDocType>,
       update: UpdateQuery<TRawDocType>,

